dist-persist/extensions/Wikibase/repo/maintenance/wbstack/EntityQuantityUnitRebuilder.php

```diff
@@ -0,0 +1,202 @@
+<?php
+
+namespace Wikibase\Repo\Maintenance;
+
+use DataValues\UnboundedQuantityValue;
+use Exception;
+use Onoi\MessageReporter\MessageReporter;
+use Wikibase\DataModel\Entity\ItemId;
+use Wikibase\DataModel\Entity\Property;
+use Wikibase\DataModel\Services\EntityId\SeekableEntityIdPager;
+use Wikibase\DataModel\Services\Lookup\ItemLookup;
+use Wikibase\DataModel\Services\Lookup\PropertyLookup;
+use Wikibase\DataModel\Services\Term\PropertyTermStoreWriter;
+use Wikibase\DataModel\Snak\PropertyNoValueSnak;
+use Wikibase\DataModel\Snak\PropertyValueSnak;
+use Wikibase\DataModel\Statement\StatementList;
+use Wikibase\Lib\Rdbms\RepoDomainDb;
+use Wikibase\Repo\WikibaseRepo;
+use Wikibase\DataModel\Entity\NumericPropertyId;
+
+/**
+ * @license GPL-2.0-or-later
+ */
+class EntityQuantityUnitRebuilder {
```

Comment:
A script like this could be useful for other Wikibase users, so it should be considered at least that this ends up in Wikibase itslef.

Reply:
+1, started working on integrating it into Wikibase + unit test here https://github.com/deer-wmde/Wikibase/tree/rebuildEntityQuantityUnit

will eventually end up on gerrit I hope - not sure if we'll merge this PR or just use it's contents like that - I'll convert it to a draft for now

Reply:
closing this now as it's now on gerrit https://gerrit.wikimedia.org/r/c/mediawiki/extensions/Wikibase/+/833742/